src/Compilers/CSharp/Portable/Symbols/Symbol_Attributes.cs

```diff
@@ -353,14 +353,7 @@ internal virtual void PostDecodeWellKnownAttributes(ImmutableArray<CSharpAttribu
             if (lazyCustomAttributesBag.SetAttributes(boundAttributes))
             {
                 this.RecordPresenceOfBadAttributes(boundAttributes);
-                if (addToDiagnostics == null)
-                {
-                    this.AddDeclarationDiagnostics(diagnostics);
-                }
-                else
-                {
-                    addToDiagnostics.AddRange(diagnostics);
-                }
+                declarationDiagnostics.AddRange(diagnostics);
```

Comment:
AddDeclarationDiagnostics is currently on Symbol. I could push down the property I introduced (DeclarationDiagnostics) to Symbol, if you prefer. Then AddDeclarationDiagnostics could use that property.